packages/node/src/indexer/fetch.service.ts

```diff
@@ -481,23 +493,45 @@ export class FetchService implements OnApplicationShutdown {
           this.eventEmitter.emit(IndexerEvent.SkipDictionary);
         }
       }
-
       const endHeight = this.nextEndBlockHeight(
         startBlockHeight,
         scaledBatchSize,
       );
 
       if (handlers.length && this.getModulos().length === handlers.length) {
         this.blockDispatcher.enqueueBlocks(
-          this.getEnqueuedModuloBlocks(startBlockHeight),
+          this.filteredBlockBatch(
+            this.getEnqueuedModuloBlocks(startBlockHeight),
+          ),
         );
       } else {
         this.blockDispatcher.enqueueBlocks(
-          range(startBlockHeight, endHeight + 1),
+          this.filteredBlockBatch(range(startBlockHeight, endHeight + 1)),
         );
       }
     }
   }
+  private filteredBlockBatch(batchBlocks: number[]): number[] {
+    const minBypass = Math.min(...this.bypassBlocks);
+    const maxBatchBlock = Math.max(...batchBlocks);
+    const bypassingBlocks = batchBlocks.filter((blk) =>
+      this.bypassBlocks.includes(blk),
+    );
+
+    if (!this.bypassBlocks?.length && maxBatchBlock < minBypass) {
```

Comment:
is this suppose be or condition? 
if this.bypassBlocks is empty we can return, or if maxBatchBlock < minBypass we also return? 
Indeed if this.bypassBlocks is empty, minBypass will return Infinity, so check 
this.bypassBlocks?.length is not necessary here.